homeassistant/components/sensor/nzbget.py

```diff
@@ -0,0 +1,147 @@
+"""
+Support for monitoring NZBGet nzb client.
+
+Uses NZBGet's JSON-RPC API to query for monitored variables.
+"""
+import logging
+from datetime import timedelta
+import requests
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = []
+SENSOR_TYPES = {
+    "ArticleCacheMB": ("Article Cache", "MB"),
+    "AverageDownloadRate": ("Average Speed", "MB/s"),
+    "DownloadRate": ("Speed", "MB/s"),
+    "DownloadPaused": ("Download Paused", None),
+    "FreeDiskSpaceMB": ("Disk Free", "MB"),
+    "PostPaused": ("Post Processing Paused", None),
+    "RemainingSizeMB": ("Queue Size", "MB"),
+}
+DEFAULT_TYPES = [
+    "DownloadRate",
+    "DownloadPaused",
+    "RemainingSizeMB",
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up nzbget sensors."""
+    base_url = config.get("base_url")
+    name = config.get("name", "NZBGet")
+    username = config.get("username")
+    password = config.get("password")
+    monitored_types = config.get("monitored_variables", DEFAULT_TYPES)
+
+    if not base_url:
+        _LOGGER.error("Missing base_url config for NzbGet")
+        return False
+
+    url = "{}/jsonrpc".format(base_url)
+
+    nzbgetapi = NZBGetAPI(api_url=url, username=username, password=password)
```

Comment:
Can you validate that the credentails are correct here and return `False` if not?